runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -60,12 +61,24 @@ public class Table<K extends Message, V extends Message, M extends Message> impl
     // gets block on parallel stream, because the pool is exhausted with threads that are trying to acquire the VLO
     // look, which creates a circular dependency. In other words, a deadlock.
 
-    protected static final ForkJoinPool pool = new ForkJoinPool(Math.max(Runtime.getRuntime().availableProcessors() / 2, 1),
+    private static final int FJP_PARALLELISM = Math.max(Runtime.getRuntime().availableProcessors() / 2, 1);
+
+    protected static final ForkJoinPool pool = new ForkJoinPool(
+            FJP_PARALLELISM,
             pool -> {
                 final ForkJoinWorkerThread worker = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool);
                 worker.setName("Table-Forkjoin-pool-" + worker.getPoolIndex());
                 return worker;
-            }, null, true);
+            },
+            null,
+            true,
+            2 * FJP_PARALLELISM, // corePoolSize == 2 * parallelism to avoid bug JDK-8330017
+            Integer.MAX_VALUE, // default to FJP.MAX_CAP
```

Comment:
If possible could you also please have a comment on _why_ we want to set this to Integer.MAX_VALUE and what happens if the default of ForkJoinPool's MAX_CAP is left in?

Reply:
It's set to max integer to use the default value. See the FJP doc
`Using a value (for example {@code Integer.MAX_VALUE}) larger than the implementation's total thread limit has the same effect as using this limit (which is the default)`